dali/operators/generic/one_hot.h

```diff
@@ -31,17 +31,21 @@ namespace detail {
 template<typename Out, typename In>
 void DoOneHot(kernels::OutTensorCPU<Out, DynamicDimensions> output,
               kernels::InTensorCPU<In, DynamicDimensions> input, int num_classes,
-              same_as_t<Out> on_value, same_as_t<Out> off_value, int ndims, int axis = 0) {
+              same_as_t<Out> on_value, same_as_t<Out> off_value, int axis) {
   auto in = input.data;
   auto out = output.data;
-  for (int i = 0; i < volume(output.shape); ++i) out[i] = off_value;
-  int64_t inner = volume(input.shape.last(ndims - axis));
-  for (int64_t i = 0, n = input.num_elements(); i < n; i++) {
-    int cls = in[i];
-    if (cls < 0 || cls >= num_classes)
-      continue;
-    int64_t out_idx = (i % inner) + cls * inner + (i / inner) * (inner * num_classes);
-    out[out_idx] = on_value;
+  auto volume_outer = volume(output.shape.begin(), output.shape.begin() + axis);
+  auto volume_inner = volume(output.shape.begin() + axis + 1, output.shape.end());
+  for (int i = 0; i < volume(output.shape); ++i) {
+    out[i] = off_value;
+  }
+  for (int64_t outer_coord = 0; outer_coord < volume_outer; outer_coord++) {
```

Comment:
As I understand it is not one hot anymore, as per each sample you may have more than one element in the whole tensor set. Does it still make sense?

Reply:
It's as the doc says. We produce output with additional axis. For any (input) coordinate that has there a value `cls` we put the on_value in the extended coordinate.
If the coordinate is `x0, x1, ..., xn`, for `axis=-1` we would have on_value in `x0, x1, ..., xn, cls`; for axis=1, it's `x0, cls, x1, ... xn` etc output[x0, j, x1, ... xn]` for j != cls will have off_value.